obf-generation/travel-gpx/README.md

```diff
@@ -0,0 +1,144 @@
+# TravelGpx GPX<->OBF tags schema (2024/11/29)
+### Generation: OsmGpxWriteContext, RouteRelationExtractor
+### Application: TravelObfHelper, TravelGpx, resources, etc
+
+## 1. Notes
+
+```
+OBF - located in OBF file
+GPX - located in GPX file
+POI - located in poi_types.xml
+MAP - located in rendering_types.xml
+STYLE - located in routes.addon.render.xml
+FLEXIBLE - not hardcoded in Java (**prefix-based**)
+METADATA - must be stored in GPX metadata extensions
+SPECIAL - tag is used/generated/transformed in a special way
+```
+
+## 2. Save/Restore (OsmAnd GPX appearance tags)
+
+### 2.1. Track
+
+```
+GPX=osmand:color OBF=color POI,MAP,STYLE,SPECIAL
+GPX=osmand:width OBF=gpx_width POI,MAP,STYLE
+
+GPX=osmand:activity OBF=gpx_osmand:activity POI,METADATA
+
+GPX=osmand:points_groups OBF=points_groups_names POI
+GPX=osmand:points_groups OBF=points_groups_icons POI
+GPX=osmand:points_groups OBF=points_groups_colors POI
+GPX=osmand:points_groups OBF=points_groups_backgrounds POI
+
+GPX=osmand:coloring_type OBF=gpx_coloring_type POI,FLEXIBLE
+GPX=osmand:color_palette OBF=gpx_color_palette POI,FLEXIBLE
+GPX=osmand:elevation_meters OBF=gpx_elevation_meters POI,FLEXIBLE
+GPX=osmand:line_3d_visualization_by_type OBF=gpx_line_3d_visualization_by_type POI,FLEXIBLE
+GPX=osmand:line_3d_visualization_position_type OBF=gpx_line_3d_visualization_position_type POI,FLEXIBLE
+GPX=osmand:line_3d_visualization_wall_color_type OBF=gpx_line_3d_visualization_wall_color_type POI,FLEXIBLE
+GPX=osmand:show_arrows OBF=gpx_show_arrows POI,FLEXIBLE
+GPX=osmand:show_start_finish OBF=gpx_show_start_finish POI,FLEXIBLE
+GPX=osmand:split_interval OBF=gpx_split_interval POI,FLEXIBLE
+GPX=osmand:split_type OBF=gpx_split_type POI,FLEXIBLE
+GPX=osmand:vertical_exaggeration_scale OBF=gpx_vertical_exaggeration_scale POI,FLEXIBLE
+```
+
+### 2.2. Waypoints
+
+```
+GPX=osmand:color OBF=color POI,MAP,STYLE,SPECIAL
+GPX=osmand:icon OBF=gpx_icon POI,STYLE
+GPX=osmand:background OBF=gpx_bg POI,STYLE
+GPX=<type> OBF=points_groups_category POI
+```
+
+## 3. Save/Restore (shields and OSM-related tags)
+
+```
+GPX=osmand:osm_tag_* OBF=[POI|MAP] POI,MAP,STYLE,FLEXIBLE (some OSM tags will be lost: osmc:symbol, route, type, etc)
+
+GPX=osmand:use_osmc_colors OBF=use_osmc_colors POI,MAP,STYLE
+
+GPX=osmand:osm_id OBF=osm_id POI,MAP (will be used to keep/display OSM id if applicable)
```

Comment:
If we use id to identify uniquely to restore we need to call it osmand:route_id. In case it's missing we need to document how it will be generated

Reply:
osm_id removed, route_id improved